rozenborg/test-simulations-mgmt-repo
# Delivering output for all task

#### What does this PR do?
Complete outputs for
1. Automating task
2. Checking repository health
3. Delivering product
4. Getting peer reviews
5. Reflecting on the learning experience
6. Setting up continuous integration
7. Setting up linters
8. Setting up software deployment environments
9. Setting up repository
10. Solving the problem
11. Using design patterns
12. Writing html, css, and js outputs
13. Writing tests.

#### Description of Task to be completed?
**1. Automating task** A screencast showing gulp as my task runner, restarting my server, with a live reload of the browser when task are made to html, css, and js, files.
**2. Checking repository health** A screenshot showing coveralls test coverage badge and codeclimate GPA on my repository's readme.
**3. Delivering product*** A google document showing product release note
**4. Getting peer reviews:** a Pull Request showing at least two Peer Code Reviews
**5. Reflecting on the learning experience*** A [link](https://medium.com/@ajudensi/reflecting-on-my-learning-experience-completing-checkpoint-i-54f3c3a581bd) to a medium post reflecting on my learning experience
**6. Setting up continuous integration** a link to a successful TravisCI build of your development branch with test runs.
7. **Setting up linters** two screenshots of ESLint Airbnb style violations in the terminal and HoundCI comment on a PR.
**8. Setting up software deployment environments** live URLs to a review app, your staging & production apps.
**9. Setting up repository**  deliver the repository for the project and at least one Pull Request.
**10. Solving the problem** deliver the file containing the implementation including all relevant dependencies. (e.g. InvertedIndex class).
**11 Using design patterns**
**12. Writing html, css, and js outputs** a folder containing the template for the proposed frontend of your project.
**13 Writing tests.**  test files for the Inverted Index class and all associated dependencies.

#### How should this be manually tested?
**1. Automating task** [Automating task 1](https://www.youtube.com/watch?v=pVn080h3Y0o),
 [Automating task 2](https://www.youtube.com/watch?v=8RpK1UMfzE4)
**2. Checking repository health** [Link](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-judensi/06-building-a-program-start-to-finish/checking-repository-health/output)
**3. Delivering product** [Release note](https://docs.google.com/document/d/1FRO-L9GTJ2SQ4wiRNl1l2nPUUEtYjAQG7kkKLLXewVI/edit)
**4. Getting peer reviews:** (Peer reviews)[https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-judensi/06-building-a-program-start-to-finish/getting-peer-reviews/output]
**5. Reflecting on learning experience** [Medium post](https://medium.com/@ajudensi/reflecting-on-my-learning-experience-completing-checkpoint-i-54f3c3a581bd)
**6. Setting up continuous integration** [Travis build](https://travis-ci.org/Andela-JUdensi/Inverted-Index/jobs/216849753)
**7. Setting up linters** [Link](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-judensi/06-building-a-program-start-to-finish/setting-up-linters/output)
**8. Setting up software deployment environments** [Review app](https://idexii-review.herokuapp.com/), [Staging app](http://idexii-staging.herokuapp.com/), [Production app](http://idexii.herokuapp.com/) 
**9. Setting up repository**  [Repository link](https://github.com/Andela-JUdensi/Inverted-Index), with a comprehensive README.md file.
**10. Solving the problem** [InvertedIndex](https://github.com/andela/test-simulations-mgmt-repo/blob/bayo-kakashi-judensi/06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js) with a descriptive JSDOC for each method and property.
**11. Using design patterns** [Link](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-judensi/06-building-a-program-start-to-finish/using-design-patterns/output)
**12. Writing html, css, and js outputs:**  [Link](https://github.com/andela/test-simulations-mgmt-repo/tree/bayo-kakashi-judensi/06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/output)
**13 Writing tests**  [InvertedIndex class test](https://github.com/andela/test-simulations-mgmt-repo/blob/bayo-kakashi-judensi/06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js). This test uses Jasmine and Expect.js as the assertion library.


#### Any background context you want to provide?
1. Local test coverage is also provided using istanbul

#### What are the relevant pivotal tracker stories?
N/A
#### Screenshots (if appropriate)

##### Hound CI error
<img width="811" alt="screen shot 2017-03-30 at 1 14 53 pm" src="https://cloud.githubusercontent.com/assets/26272640/24553265/965c01aa-1621-11e7-93d2-56a1eb37b262.png">

##### Eslint errors
<img width="644" alt="screen shot 2017-03-30 at 1 32 01 pm" src="https://cloud.githubusercontent.com/assets/26272640/24553285/aef1305a-1621-11e7-8cdc-7b97b0a4ff47.png">

##### Codeclimate and CI badges
<img width="620" alt="screen shot 2017-03-30 at 1 37 40 pm" src="https://cloud.githubusercontent.com/assets/26272640/24553304/c2cc3ba6-1621-11e7-9657-a7de38deac5d.png">

#####  Successful Travis Build
<img width="932" alt="screen shot 2017-03-30 at 1 58 13 pm" src="https://cloud.githubusercontent.com/assets/26272640/24553329/d8c27d94-1621-11e7-8334-daf65b25698d.png">

##### IDex - Application interface
<img width="1278" alt="screen shot 2017-03-31 at 1 18 45 pm" src="https://cloud.githubusercontent.com/assets/26272640/24553354/ea7b9ad4-1621-11e7-804f-f9b4dfb49897.png">

##### iDex - Application interface with index created
<img width="1278" alt="screen shot 2017-03-31 at 2 54 59 pm" src="https://cloud.githubusercontent.com/assets/26272640/24553397/0fb31016-1622-11e7-811b-d3f92bceff37.png">

##### iDex - Application interface with search result for multiple files
<img width="1280" alt="screen shot 2017-03-31 at 2 57 12 pm" src="https://cloud.githubusercontent.com/assets/26272640/24553521/5c9e4d96-1622-11e7-82ad-be811d1e1482.png">


#### Questions:
N/A
